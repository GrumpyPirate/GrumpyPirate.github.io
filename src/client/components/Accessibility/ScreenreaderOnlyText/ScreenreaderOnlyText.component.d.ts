interface ScreenreaderOnlyTextProps {
  text: string;
}

export { ScreenreaderOnlyTextProps };
