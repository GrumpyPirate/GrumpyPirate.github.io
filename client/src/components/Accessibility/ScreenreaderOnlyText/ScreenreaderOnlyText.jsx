import React from 'react';
import PropTypes from 'prop-types';

import classes from './ScreenreaderOnlyText.scss';

const ScreenreaderOnlyText = ({ text }) => (
  <span className={classes['screenreader-only']}>{text}</span>
);

ScreenreaderOnlyText.propTypes = {
  text: PropTypes.string.isRequired,
};

export default ScreenreaderOnlyText;
