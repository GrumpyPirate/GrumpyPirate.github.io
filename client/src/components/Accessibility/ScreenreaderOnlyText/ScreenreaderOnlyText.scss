.screenreader-only {
  position: absolute;
  left: -200vw;
  width: 1px;
  height: 1px;
  clip: rect(0, 0, 0, 0);
  overflow: hidden;
}
