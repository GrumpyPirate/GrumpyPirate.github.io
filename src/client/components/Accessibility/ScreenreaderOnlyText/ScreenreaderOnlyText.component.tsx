import React, { FunctionComponent } from 'react';

import { ScreenreaderOnlyTextProps } from './ScreenreaderOnlyText.component.d';
import classes from './ScreenreaderOnlyText.component.scss';

const ScreenreaderOnlyText: FunctionComponent<ScreenreaderOnlyTextProps> = ({
  text,
}: ScreenreaderOnlyTextProps) => <span className={classes['screenreader-only']}>{text}</span>;

export default ScreenreaderOnlyText;
