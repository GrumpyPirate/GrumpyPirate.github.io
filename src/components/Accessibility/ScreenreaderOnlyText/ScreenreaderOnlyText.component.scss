.screenreader-only {
  clip: rect(0, 0, 0, 0);
  height: 1px;
  left: -200vw;
  overflow: hidden;
  position: absolute;
  width: 1px;
}
