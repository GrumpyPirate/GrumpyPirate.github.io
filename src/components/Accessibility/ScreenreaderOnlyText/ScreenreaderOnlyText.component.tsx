import React, { FunctionComponent } from 'react';

import { ScreenreaderOnlyTextProps } from './ScreenreaderOnlyText.component.types';
import classes from './ScreenreaderOnlyText.component.scss';

const ScreenreaderOnlyText: FunctionComponent<ScreenreaderOnlyTextProps> = ({
  text,
}: ScreenreaderOnlyTextProps) => <span className={classes['screenreader-only']}>{text}</span>;

export default ScreenreaderOnlyText;
