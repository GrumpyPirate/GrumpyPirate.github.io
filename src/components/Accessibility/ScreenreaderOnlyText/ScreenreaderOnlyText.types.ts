import { PropsWithClassName } from 'types/common';

export type ScreenreaderOnlyTextProps = PropsWithClassName<{
  text: string;
}>;
