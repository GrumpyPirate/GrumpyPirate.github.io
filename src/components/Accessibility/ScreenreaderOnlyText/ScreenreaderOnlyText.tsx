import React, { FunctionComponent } from 'react';
import styled from 'styled-components';

import { visuallyHidden } from 'styles';

import { ScreenreaderOnlyTextProps } from './ScreenreaderOnlyText.types';

const ScreenreaderOnlyText: FunctionComponent<ScreenreaderOnlyTextProps> = function ({
  className,
  text,
}) {
  return <span className={className}>{text}</span>;
};

export default styled(ScreenreaderOnlyText)`
  ${visuallyHidden}
`;
