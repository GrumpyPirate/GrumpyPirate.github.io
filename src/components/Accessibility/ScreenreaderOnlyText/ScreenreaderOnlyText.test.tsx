import { render } from '@testing-library/react';
import React from 'react';

import ScreenreaderOnlyText from './ScreenreaderOnlyText';
import fixture from './ScreenreaderOnlyText.fixture';

describe('Components', () => {
  describe('ScreenreaderOnlyText', () => {
    const props = {
      ...fixture,
    };

    it('should render as expected, without crashing', () => {
      const { container } = render(<ScreenreaderOnlyText {...props} />);

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
