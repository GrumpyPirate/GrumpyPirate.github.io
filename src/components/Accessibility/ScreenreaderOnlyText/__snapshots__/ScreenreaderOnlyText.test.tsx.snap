// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components ScreenreaderOnlyText should render as expected, without crashing 1`] = `
.c0 {
  position: absolute;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  padding: 0;
  border: 0;
  height: 1px;
  width: 1px;
  overflow: hidden;
}

<span
  class="c0"
>
  Nobis doloribus quidem
</span>
`;
